dot/sync/chain_sync.go

```diff
@@ -932,6 +936,34 @@ func (cs *chainSync) validateJustification(bd *types.BlockData) error {
 	return nil
 }
 
+func (cs *chainSync) getHighestBlock() (int64, error) {
+	cs.RLock()
+	ps := cs.peerState
+	cs.RUnlock()
```

Comment:
```suggestion
	cs.RLock()
	defer cs.RUnlock()
	ps := cs.peerState
```

afaik `ps` will still point to `cs.peerState`, it's just an alias. so you'd need to unlock when you're completely done with reading the map, or there could be a race condition

Reply:
I'd also probably remove the `ps := cs.peerState` as well as just reference `cs.peerState` directly

Reply:
Its true @noot. I checked it here: https://go.dev/play/p/bCsiQceROqb

I didn't notice that all values in the map has the same reference... Good catch.

Reply:
Done in 60eaaa8